src/core/ext/filters/client_channel/resolver/binder/binder_resolver.cc

```diff
@@ -95,10 +89,10 @@ class BinderResolverFactory : public ResolverFactory {
     static_assert(sizeof(un->sun_path) >= 101,
                   "unix socket path size is unexpectedly short");
     if (path.size() + 1 > sizeof(un->sun_path)) {
-      return GRPC_ERROR_CREATE_FROM_CPP_STRING(path +
-                                               " is too long to be handled");
+      return GRPC_ERROR_CREATE_FROM_CPP_STRING(
+          absl::StrCat(path, " is too long to be handled"));
     }
-    strcpy(un->sun_path, path.c_str());
+    strcpy(un->sun_path, std::string(path).c_str());
```

Comment:
There's no need to create a `std::string` here; that creates an unnecessary temporary copy.  Instead, just do:

```
memcpy(un->sun_path, path.data(), path.size());
un->sun_path[path.size()] = '\0';
```

Reply:
Actually, to be safe, let's just `memset()` the entire `grpc_resolved_address` to all-zero before you populate the string.  Then there's no need to explicitly set the trailing nul byte here.

Reply:
Oh, and I just noticed that you're already doing that.  So nevermind, just the `memcpy()` should be sufficient here.